app/code/Magento/MediaGalleryUi/Model/GetAssetUsageDetails.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryUi\Model;
+
+use Magento\Backend\Model\UrlInterface;
+use Magento\Framework\Exception\IntegrationException;
+use Magento\MediaContentApi\Api\GetContentByAssetIdsInterface;
+
+/**
+ * Provide information on which content asset is used in
+ */
+class GetAssetUsageDetails
+{
+    /**
+     * @var GetContentByAssetIdsInterface
+     */
+    private $getContent;
+
+    /**
+     * @var UrlInterface
+     */
+    private $url;
+
+    /**
+     * @var array
+     */
+    private $contentTypes;
+
+    /**
+     * @param GetContentByAssetIdsInterface $getContent
+     * @param UrlInterface $url
+     * @param array $contentTypes
+     */
+    public function __construct(
+        GetContentByAssetIdsInterface $getContent,
+        UrlInterface $url,
+        array $contentTypes = []
+    ) {
+        $this->getContent = $getContent;
+        $this->url = $url;
+        $this->contentTypes = $contentTypes;
+    }
+
+    /**
+     * Provide information on which content asset is used in
+     *
+     * @param int $id
+     * @return array
+     * @throws IntegrationException
+     */
+    public function execute(int $id): array
+    {
+        $details = [];
+
+        foreach ($this->getUsageByEntities($id) as $type => $entities) {
+            $details[] = [
+                'name' => $this->getName($type),
+                'number' => count($entities),
+                'link' => $this->getLinkUrl($type)
+            ];
+        }
+
+        return $details;
```

Comment:
How about using `array_map` for that purpose?

Reply:
I would love to use it, but `array_map` does not provide the callback with a key

Reply:
OK